chalice/local.py

```diff
@@ -201,7 +201,7 @@ class NotAuthorizedError(LocalGatewayException):
 
 
 class NoOptionsRouteDefined(LocalGatewayException):
-    CODE = 403
+    CODE = 200
```

Comment:
Why was this changed?

Reply:
Because thats what API gateway does. I missed it before but I noticed it when going over these the tests for cors cases. I can do this as a separate PR if you would prefer.

Reply:
I don't see a test for it.  I don't mind it in this PR, but we just have a test for this.  Might also be good to have a separate commit for this with a description of why it's being changed.

Reply:
The existing tests should have covered it and I corrected the tests while I was there anyway and thats when I noticed it was incorrect.

I added these assertions:
* https://github.com/stealthycoin/chalice/blob/52f48152c7d2206c6fe9fe62cc39cf243f52e9fa/tests/unit/test_local.py#L348

* https://github.com/stealthycoin/chalice/blob/52f48152c7d2206c6fe9fe62cc39cf243f52e9fa/tests/unit/test_local.py#L364

Reply:
I don't follow.  This is updating the code for an exception right?  Those tests are for successful requests aren't they?  Are you saying we were previously returning a 403 response when we should have been returning a 200?

Reply:
The exception is raised when there is no OPTIONS method defined on a route which LocalGateway exceptions are raised when the routes cannot process the request, so that is reasonable so far. The calling code can still generate the OPTIONS request that API Gateway would return and then send a successful response if there is a CORSConfig object on it. I missed that it was a success case somehow before. But looking at it again it seems like it should be decoupled entirely from that type of error since all the others exceptions like this are actual error cases. Whereas this one should be caught and turned into a successful response by auto-generating the CORS headers.

I will fix this in a separate PR.